openedx/tests/xblock_integration/xblock_testcase.py

```diff
@@ -0,0 +1,202 @@
+        )
+
+        cls.chapter = ItemFactory.create(
+            parent=cls.course,
+            display_name='Overview',
+            category='chapter'
+        )
+
+        cls.urls = {}
+
+        with cls.store.bulk_operations(cls.course.id, emit_signals=False):
+            for section in cls.test_configuration:
+                section_item = ItemFactory.create(
+                    parent=cls.chapter,
+                    display_name=section['urlname'],
+                    category='sequential'
+                )
+
+                cls.urls[section['urlname']] = reverse(
+                    'courseware_section',
+                    kwargs={
+                        'course_id': unicode(cls.course.id),
+                        'chapter': 'Overview',
+                        'section': section['urlname'],
+                    }
+                )
+                print section['urlname']
+                print cls.urls[section['urlname']]
+
+                blocks = section["xblocks"]
+
+                unit_item = ItemFactory.create(
+                    parent=section_item,
+                    display_name="unit_"+section['urlname'],
+                    category='vertical'
+                )
+                for block in blocks:
+                    block_item = ItemFactory.create(
+                        parent=unit_item,
+                        category=block['blocktype'],
+                        display_name=block['urlname']
+                    )
+            cls.course_url = reverse(
+                'courseware_section',
+                kwargs={
+                    'course_id': unicode(cls.course.id),
+                    'chapter': 'Overview',
+                    'section': section['urlname'],
+                }
+            )
+            print cls.course_url
+
+            cls.course_url = reverse(
+                'courseware_section',
+                kwargs={
+                    'course_id': unicode(cls.course.id),
+                    'chapter': u'Overview',
+                    'section': u'two_done_block_test',
+                }
+            )
+            print cls.course_url
+
+    def _containing_section(self, block_urlname):
+        '''
+        For a given block, return the parent section
+        '''
+        for section in self.test_configuration:
+            blocks = section["xblocks"]
+            for block in blocks:
+                if block['urlname'] == block_urlname:
+                    return section['urlname']
+        raise Exception("Block not found "+block_urlname)
+
+    def assertXBlockScreenshot(self, block_urlname, rendering=None):
+        '''
+        As in Bok Choi, but instead of a CSS selector, we pass a
+        block_id. We may want to be able to pass an optional selector
+        for picking a subelement of the block.
+
+        This confirms status code, and that the screenshot is
+        identical.
+
+        TODO: IMPLEMENT
+        '''
+        pass
```

Comment:
This actually seems likely to be fragile to me. The XBlock won't be defining all of its styling, so screenshots will vary based on the Runtime being used. Visual Diffing works better on an ongoing basis (to verify that the visual contents are changing from one commit to another), but that would be on a per-runtime basis.

Given that, I'd rather see the tests using bok_choy style functionality tests that verify the behavior of the XBlocks.